Point cloud editing, show human readable values

uclaros opened the pull request:
## Description
Use the spinbox's suffix to display human readable representation of current value.
Applies for Classification and boolean attributes.

![image](https://github.com/user-attachments/assets/ce24e356-63b6-4548-a830-a030c30b22ca)

Also fixes the ScanAngleRank editing range to be between -180 and 180 degrees, which is the values we return when identifying the points.
<!--
  BEFORE HITTING SUBMIT -- Please BUILD AND TEST your changes thoroughly. This is YOUR responsibility! Do NOT rely on the QGIS code maintainers to do this for you!!

  IMPORTANT NOTES FOR FIRST TIME CONTRIBUTORS
  ===========================================

  Congratulations, you are about to make a pull request to QGIS! To make this as easy and pleasurable for everyone, please take the time to read these lines before opening the pull request.

  Include a few sentences describing the overall goals for this pull request (PR). If applicable also add screenshots or - even better - screencasts.
  Include both: *what* you changed and *why* you changed it.

  If this is a pull request that adds new functionality which needs documentation, give an especially detailed explanation.
  In this case, start with a short abstract and then write some text that can be copied 1:1 to the documentation in the best case.

  Also mention if you think this PR needs to be backported. And list relevant or fixed issues.

------------------------

  Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by checking the following list.
  Feel free to ask in a comment if you have troubles with any of them.

  - Commit messages are descriptive and explain the rationale for changes.

  - Commits which fix bugs include `Fixes #11111` at the bottom of the commit message. If this is your first pull request and you forgot to do this, write the same statement into this text field with the pull request description.

  - New unit tests have been added for relevant changes

  - you have read the QGIS Developer Guide (https://docs.qgis.org/testing/en/docs/developers_guide/index.html) and your PR complies with its QGIS Coding Standards
-->

uclaros committed:
Use spinbox suffix to display the human readable current value

uclaros committed:
Use -180 to 180 range for ScanAngleRank

uclaros committed:
clang tidy

nyalldawson commented:
Wouldn't a combo-box be a more suitable widget here?

uclaros committed:
use combobox for classification and boolean attributes

uclaros commented:
> Wouldn't a combo-box be a more suitable widget here?

This was my initial intent (hinted by the branch name :) but was failing to toggle the spinbox/combobox visibility.
Turned out it was this thing with QToolBars where one needs to toggle visibility of the _QAction_ returned when adding a widget to the toolbar.

Now a combo box is used.
The logic is that it gets populated with classes from the classification renderer (if available) and actual classes present in the dataset (from the layer stats).
These are added on top for easy access, then there's a separator and all the standard las class codes follow.
![image](https://github.com/user-attachments/assets/454a18b2-d0a4-4e03-98cc-ded1292532ef)

nyalldawson commented:
Thanks @uclaros , looks great!

nyalldawson commented:
@uclaros where's the "Renamed class" string come from? (There's inconsistent capitalisation for that one)

uclaros commented:
> @uclaros where's the "Renamed class" string come from? (There's inconsistent capitalisation for that one)

That one is there to showcase that if a class is renamed in the Classification renderer available classes, this name is used instead of the default las standard class names.

uclaros committed:
Change the combobox logic

Now the combobox is populated with all classes from the Classification renderer, which may have a modified name, and
all available classes in the dataset, based on layer statistics. The combobox is editable so that the rest classes
can be typed-in.

uclaros committed:
fix qt6 build

DelazJ commented:
@uclaros may i ask you to update the PR message (if needed) and label correctly for both changelog and documentation? This would apply also to some other PRs about point cloud editing otherwise, Docs and changelog are unaware of your work...
